spec/_attachments/ic.did

```diff
@@ -170,6 +170,12 @@ service ic : {
       memory_size: nat;
       cycles: nat;
       idle_cycles_burned_per_day: nat;
+      query_stats: vec record {
```

Comment:
There are two problems here:
1. The drafted implementation in the mono-repo seems to be inconsistent with the spec here:
    - This spec defines the `query_states` to be a Vector of the Record;
    - The implementation has no outer Vector

2. The extra field `query_stats` should be optional so that it can be backward-compatible:
```
query_stats: opt record {
  ...
}
```

Reply:
Agreed with 1.

Ad 2. The new field is added to the response and thus old clients should be able to parse a response with the new field into the old type without the field.

Reply:
Yep, adding a new field in the return position is backward compatible. See Roman's explanation: https://mmapped.blog/posts/20-candid-for-engineers.html#subtyping

Reply:
"Adding a new field to a record in the method result type is backward compatible" is true for normal canister. Because the consumers always upgrade later than the provider.

But for the management canister, the case is different. Let's say, if the API in Rust CDK is updated and a canister using the new CDK is deployed to a network with old management canister API. The canister will fail to decode the return value because it expects a field which is not provided.

We already see some canister projects keep using old version of `dfx`. Such projects won't be able to upgrade their CDK unless they upgrade `dfx` at the same time.

Reply:
> The canister will fail to decode the return value because it expects a field which is not provided.
No, that's not possible. The return value from the management canister *always* contains the new field. It's up to the client to decide if they want to decode that new field.

Reply:
> The return value from the management canister always contains the new field.

This will be true for the mainnet. But for those projects using old local replica from old `dfx` in their CI, if they upgrade Rust CDK, the CI will fail. And they will have to upgrade `dfx`.

If I did complete the support in Rust CDK, I wouldn't be able to merge the PR and release today. Because the mainnet doesn't have the feature yet. All the CDK/agent supports must be held to release until the mainnet upgrades.

In other words, how do we treat the case of "new client <-> old server"?
Should we simply ignore it? 
Or if we make the new field `opt`, this case will be handled gracefully. And we completely decouple the server and clients. Server and clients can release at their own paces.

Reply:
That all seems to be expected. If they upgrade CDK, they need to upgrade dfx, and CDK changes should be done after it's been rolled out in the main net, or included in the dfx release.

> In other words, how do we treat the case of "new client <-> old server"?
Should we simply ignore it?

I think we can ignore it. It's too restrictive to require both backward and forward compatibility. OTOH, a good way to solve this problem is to make CDK to always fetch the latest did from the network, and generate the binding, instead of hardcoding this binding in a CDK release.